src/inmanta/data/__init__.py

```diff
@@ -5030,129 +5039,125 @@ def done(self) -> int:
         return self._done
 
     @classmethod
-    async def create_and_insert(
+    async def create_for_partial_compile(
         cls,
         env_id: uuid.UUID,
         version: int,
         total: int,
         version_info: Optional[JsonType],
         undeployable: abc.Sequence[ResourceIdStr],
         skipped_for_undeployable: abc.Sequence[ResourceIdStr],
-        partial_base: Optional[int] = None,
-        rids_in_partial_compile: Optional[abc.Set[ResourceIdStr]] = None,
+        partial_base: int,
+        rids_in_partial_compile: abc.Set[ResourceIdStr],
         connection: Optional[Connection] = None,
     ) -> "ConfigurationModel":
         """
-        Create a new ConfigurationModel instance and insert it into the database. When partial_base is provided,
-        the new ConfigururationModel will contain all the undeployables and skipped_for_undeployables present
-        in the partial_base version that are not part of the partial compile, i.e. not present in rids_in_partial_compile.
-        """
-        rids_in_partial_compile = rids_in_partial_compile if rids_in_partial_compile is not None else set()
-        now = datetime.datetime.now().astimezone()
-        if partial_base is None:
-            cm = ConfigurationModel(
-                environment=env_id,
-                version=version,
-                date=now,
-                total=total,
-                version_info=version_info,
-                undeployable=undeployable,
-                skipped_for_undeployable=skipped_for_undeployable,
+        Create and insert a new configurationmodel that is the result of a partial compile. The new ConfigururationModel will
+        contain all the undeployables and skipped_for_undeployables present in the partial_base version that are not part of
+        the partial compile, i.e. not present in rids_in_partial_compile.
+        """
+        query = f"""
+            WITH base_version_exists AS (
+                SELECT COUNT(*) != 0 AS base_version_found
```

Comment:
This is equivalent to `EXISTS`, isn't it?

Reply:
Yes. I turned it into an EXISTS query.